CppCustomVisualizer/dll/_EntryPoint.cpp

```diff
@@ -348,7 +350,7 @@ HRESULT CCppCustomVisualizerService::FileTimeToText(const FILETIME& fileTime, CS
         return WIN32_LAST_ERROR();
     }
 
-    pBuffer += (cch-1L); // '-1' is to convert from a character count (including null terminator) to a length
+    pBuffer += ((int64_t)cch-1); // '-1' is to convert from a character count (including null terminator) to a length
```

Comment:
>int64_t [](http://example.com/codeflow?start=17&length=7)

Shouldn't this be `DWORD`?

Reply:
pBuffer is 64 bit according to the code analysis. According to MSDN, it's 32 bit. Which is slightly confusing. 

I believe the issue is we're casting cch (4 bytes) to pBuffer (8 byte pointer), so casting cch to 8 bytes would be the fix?

Reply:
`cch` is a 32-bit integer. I am assuming code analysis is complaining because `cch` is signed?

We can't casting `cch` to pBuffer at all.

Reply:
`Arithmetic overflow: Using operator '-' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '-' to avoid overflow`

I think the issue is with pBuffer being 8 bytes and we are casting cch to 8 bytes after subtracting 1 (instead of before) to perform the operation.

Reply:
The rule is BS. But you could probably make it happy and still correct if you cast to `size_t`. `int64_t` is wrong as `pBuffer` isn't always 64-bit.